src/faebryk/libs/picker/jlcpcb/pickers.py

```diff
@@ -0,0 +1,491 @@
+import logging
+
+import faebryk.library._F as F
+from faebryk.core.core import Module
+from faebryk.library.can_query_jlcpcb_db import can_query_jlcpcb_db
+from faebryk.library.can_query_jlcpcb_db_defined import can_query_jlcpcb_db_defined
+from faebryk.libs.picker.jlcpcb.jlcpcb import (
+    JLCPCB_DB,
+    ComponentQuery,
+    MappingParameterDB,
+)
+from faebryk.libs.picker.picker import (
+    DescriptiveProperties,
+    PickError,
+    PickErrorNotImplemented,
+    has_part_picked,
+)
+from faebryk.libs.units import si_str_to_float
+
+logger = logging.getLogger(__name__)
+
+
+def pick_module_by_query(module: Module, qty: int = 1) -> None:
+    if not module.has_trait(can_query_jlcpcb_db):
+        find_query_for_module(module)
+
+    db = JLCPCB_DB()
```

Comment:
I feel like this is not a good idea to instantiate everytime

Reply:
I don't really like the alternatives I came up with:
- Instantiate it globally (it would keep on running)
- Instantiate it in the app (feels like it doesn't belong there)
I guess the most realistic option is to let the recursive picker (or at that level somewhere) set up the DB/API connections and close them after picking is done

Reply:
i fixed it with a proper at_exit